### PR TITLE
Fix address request performance

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

**Comment by SpComb** [comment]
>      1.1) Failure/Error: expect(policy).to receive(:allocate_address).with((IPAddr.new('10.81.0.2/16')..IPAddr.new('10.81.255.254/16')).to_a).and_return(IPAddr.new('10.81.100.100/16'))

This shold probably be:
```ruby
expect(policy).to receive(:allocate_address).with((IPAddr.new('10.81.0.2/16')..IPAddr.new('10.81.0.101/16')).to_a).and_return(IPAddr.new('10.81.100.100/16'))
```

**Comment by jakolehm** [comment]
@SpComb @jnummelin ping

File: app/mutations/addresses/request.rb [hunk 1]
```diff
@@ -63,10 +63,10 @@ def request_static
     def request_dynamic
       available = @pool.available_addresses
 
-      info "pool #{@pool} allocates from #{@pool.allocation_range} and has #{available.length} available addresses"
+      info "pool #{@pool} allocates from #{@pool.allocation_range} and has #{available.size} available addresses"
 
       # allocate
-      unless allocate_address = policy.allocate_address(available)
+      unless allocate_address = policy.allocate_address(available.first(100).to_a)
```
Comment (jnummelin) [inline app/mutations/addresses/request.rb]: Not really sure but some part of my brain would like to see this `available.first(100).to_a` part handled within the `policy.allocate_address`. The other part is fine with this. :)

Comment (jnummelin) [inline app/mutations/addresses/request.rb]: The other part won. I think this is good as it is. :)

**Pull request merged**